docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+# Understand Jina Recursive Document Representation
+
+In Jina, each `Document` is represented as a recursive representation (tree).
+
+A rooted recursive representation has a root node and every node has X children.
+In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
+
+![rooted-binary-tree](img/rooted-binary-tree.png)
+
+The above image illustrates a minimal document structure: A document (root node) consists of two child nodes: *chunks* and *matches*.
+`chunks`s are a sequence of `Document`s which is attached to the root document with a higher `granularity` degree. `Matches` is a sequence of `Document`s  which are semantically related to the root document.
```

Comment:
if you introduce `granularity` here, I would also introduce `adjacency`

Reply:
Furthermore either `Chunks`and `Matches` or `chunks` and `matches`

Reply:
And these `granularity` is only really used in the multimodal document type. Even though it may feel quite natural given our history of document traversal, in the current structure it is something very deep and advanced. In my opinion, that should not appear anywhere in the introduction document or only at the very end.

Reply:
There is an extra "s" after chunks and the "s" after Document is detached.

Reply:
I think you don't need to put Document and Chunks in code blocks. For example, it is not used here: https://docs.jina.ai/chapters/flow/index.html#common-design-patterns

Reply:
@JoanFM I made it on purpose to introduce `g` at chunks and `a` at matches, previously I have a section to introduce both `g` and `a`, while thought it twice think it make more sense. (since putting `granularity` together with `chunk` is more understandable, same applies for `match` and `adjacency `). I did thought about you concern, so added one line:
**NOTE: granularity and adjacency apply to both chunks and matches.** later.

Reply:
I understand, no need to confuse the user. Just felt that adjacency was not introduced here, but no big deal